### ISSUE TITLE
Script variables selectable when adding animation track but key values not editable

### ISSUE THREAD

**Issue by maiself** [godotengine/godot#75127]
### Godot version

godot 4 - e01b82857

### System information

Arch Linux, Mesa Intel® UHD Graphics (CML GT2), Vulkan Forward+

### Issue description

Animation tracks can be added pointing to unexported script variables, however their keys can not be edited in the inspector when selected. They can't be animated.

### Steps to reproduce

- add a script to a node
- add a typed variable to that script (do not export the variable)
- add an animation player and a new animation
- select 'add track' -> 'property track'
- select the node that the script was attached to
- select the unexported typed variable
- right click in the new track and select 'insert key'
- click on the new key to select it
- the property inspector shows the new key, however there is no value field to edit
- if you now edit the script to export the variable there will be a value field when selecting the key

### Minimal reproduction project

N/A

**Comment by dalexeev** [comment]
The reason is that if the property is not exported (there is no `PROPERTY_USAGE_EDITOR` flag), then it is not displayed in the editor. Here the property info is copied as is, so the property doesn't appear in the AnimationTrackKeyEdit either.

https://github.com/godotengine/godot/blob/161d028ae89acb578cfcd4f29bfe9ac205459e8b/editor/animation_track_editor.cpp#L516-L523

Probably the solution would be to add `pi.usage = PROPERTY_USAGE_DEFAULT;` after 521 line.

**Comment by maiself** [comment]
Maybe `PropertyInfo pi = hint.added_usage(PROPERTY_USAGE_EDITOR);`?

It seems to work here without `@tool` and the above line change.

**Comment by dalexeev** [comment]
> It seems to work here without `@tool` and the above line change.

No, I checked it, GDScript changes are needed. See https://github.com/dalexeev/godot/commit/bc85494a05290324c8dfd15bda3b3e2b713814a8 (this is a very crude solution as it only works for builtin types).

**Comment by maiself** [comment]
Oh, I see what happened. You are right. When I reported this I had tool mode enabled. Then made the changes to `AnimationTrackKeyEdit`. Then disabled tool mode. At this point the property inspector worked and the key already existed. But its not possible to choose the property to create a new animation track.

**Comment by dalexeev** [comment]
> Maybe `PropertyInfo pi = hint.added_usage(PROPERTY_USAGE_EDITOR);`?

I'm not sure. `_list->push_back(PropertyInfo(hint.type, PNAME("value")));` option is most likely wrong, since it removes information about the class/enum. But probably not all `PROPERTY_USAGE_*` flags make sense to copy.

**Comment by maiself** [comment]
Removing information from the property isn't good, `hint` and `hint_string` are required for some types.

> But probably not all `PROPERTY_USAGE_*` flags make sense to copy.

That's exactly what happens now in master, everything is copied. So I think adding `PROPERTY_USAGE_EDITOR` should be fine, looking over the usage flags nothing jumps out at me as problematic.

**Comment by dalexeev** [comment]
> So I think adding `PROPERTY_USAGE_EDITOR` should be fine

Perhaps we should add some other flags (`PROPERTY_USAGE_STORAGE`?) or remove some of the existing ones (`PROPERTY_USAGE_CHECKABLE`, `PROPERTY_USAGE_CHECKED`, `PROPERTY_USAGE_READ_ONLY`?).

> Removing information from the property isn't good, `hint` and `hint_string` are required for some types.

Note that other condition branches of this method may also have lost some of the `PropertyInfo` data.
